openapi/openapi.json

```diff
@@ -27048,7 +27048,7 @@
           "x-kubernetes-list-type": "map"
         },
         "customLogoFile": {
-          "description": "customLogoFile replaces the default OpenShift logo in the masthead and about dialog. It is a reference to a ConfigMap in the openshift-config namespace. This can be created with a command like 'oc create configmap custom-logo --from-file=/path/to/file -n openshift-config'. Image size must be less than 1 MB due to constraints on the ConfigMap size. The ConfigMap key should include a file extension so that the console serves the file with the correct MIME type. Recommended logo specifications: Dimensions: Max height of 68px and max width of 200px SVG format preferred",
+          "description": "customLogoFile replaces the default OpenShift logo in the masthead and about dialog. It is a reference to a ConfigMap in the openshift-config namespace. This can be created with a command like 'oc create configmap custom-logo --from-file=/path/to/file -n openshift-config'. Image size must be less than 1 MB due to constraints on the ConfigMap size. The ConfigMap key should include a file extension so that the console serves the file with the correct MIME type. Recommended logo specifications: Dimensions: Max height of 60px and max width of 200px SVG format preferred Deprecated: Use Logos instead.",
```

Comment:
Do we want to validate the co-dependency between these fields using CEL? Only one of them being non-empty is possible

Reply:
> Do we want to validate the co-dependency between these fields using CEL? Only one of them being non-empty is possible

@JoelSpeed This would be good to have. Is there any example of a similar validation you could point me to?
cc: @jhadvig

Reply:
You'll have something along the lines of 
```
+kubebuilder:validation:XValidation:rule="!has(self.customLogoOld) || !has(self.customLogoNew) || self.customLogoNew.size() == 0",message="<write something meaningful here>"
```

Reply:
> You'll have something along the lines of
> 
> ```
> +kubebuilder:validation:XValidation:rule="!has(self.customLogoOld) || !has(self.customLogoNew) || self.customLogoNew.size() == 0",message="<write something meaningful here>"
> ```
What about something like this:
```
// Ensure that Logos and CustomLogoFile cannot be set at the same time.
// +kubebuilder:validation:XValidation:rule="!(has(self.logos) && has(self.customLogoFile))",message="Only one of Logos or CustomLogoFile can be set."
```

Reply:
That works, but use the serialised version of the field names in the error message please

Also, if we change the format because of discoverability (I have a thread on this) we will need to change the validation to check for a non-empty list

Reply:
> That works, but use the serialised version of the field names in the error message please
> 
> Also, if we change the format because of discoverability (I have a thread on this) we will need to change the validation to check for a non-empty list

Not sure of how to address this and the other [ comment](https://github.com/openshift/api/pull/2177#discussion_r1984823159). 

Maybe we keep as `omitempty` and the `minItems:=1`.

Reply:
How do you mean?

Can you explain the  behaviour in each of the following states, and which you'd expect to be an error, and why
```
A

customLogoFile: ""
logos: []
```
```
B

customLogoFile: foo
logos: []
```
```
C

customLogoFile: ""
logos:
- type: Masthead
  ...
```
```
D

customLogoFile: foo
logos:
- type: Masthead
  ...
```